README.md
# tocBackend

CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### [3.1.30](https://github.com/solucionesit365/tocBackend/compare/v3.1.29...v3.1.30) (2022-10-12)

### [3.1.29](https://github.com/solucionesit365/tocBackend/compare/v3.1.28...v3.1.29) (2022-10-12)

### [3.1.28](https://github.com/solucionesit365/tocBackend/compare/v3.1.27...v3.1.28) (2022-10-12)

### [3.1.27](https://github.com/solucionesit365/tocBackend/compare/v3.1.26...v3.1.27) (2022-10-12)

### [3.1.26](https://github.com/solucionesit365/tocBackend/compare/v3.1.25...v3.1.26) (2022-10-12)

### [3.1.25](https://github.com/solucionesit365/tocBackend/compare/v3.1.24...v3.1.25) (2022-09-14)

### [3.1.24](https://github.com/solucionesit365/tocBackend/compare/v3.1.23...v3.1.24) (2022-08-24)

### [3.1.23](https://github.com/solucionesit365/tocBackend/compare/v3.1.22...v3.1.23) (2022-08-22)

### [3.1.22](https://github.com/solucionesit365/tocBackend/compare/v3.1.21...v3.1.22) (2022-08-10)

### [3.1.21](https://github.com/solucionesit365/tocBackend/compare/v3.1.20...v3.1.21) (2022-08-08)

### [3.1.20](https://github.com/solucionesit365/tocBackend/compare/v3.1.19...v3.1.20) (2022-07-31)

### [3.1.19](https://github.com/solucionesit365/tocBackend/compare/v3.1.18...v3.1.19) (2022-07-30)

### [3.1.18](https://github.com/solucionesit365/tocBackend/compare/v3.1.17...v3.1.18) (2022-07-28)

### 3.1.17 (2022-07-26)

### [3.1.16](https://github.com/solucionesit365/tocBackend/compare/v3.1.15...v3.1.16) (2022-07-20)

### [3.1.15](https://github.com/solucionesit365/tocBackend/compare/v3.1.11...v3.1.15) (2022-07-14)

### [3.1.14](https://github.com/solucionesit365/tocBackend/compare/v3.1.13...v3.1.14) (2022-06-07)

### 3.1.13 (2022-05-26)

### [3.1.12](https://github.com/solucionesit365/tocBackend/compare/v3.1.9...v3.1.12) (2022-05-19)

### [3.1.11](https://github.com/solucionesit365/tocBackend/compare/v3.1.10...v3.1.11) (2022-05-05)

### 3.1.10 (2022-05-02)

### 3.1.9 (2022-04-14)

### [3.1.8](https://github.com/HitSystems/tocBackend/compare/v3.1.7...v3.1.8) (2022-04-10)

### [3.1.7](https://github.com/HitSystems/tocBackend/compare/v3.1.6...v3.1.7) (2022-04-06)

### [3.1.6](https://github.com/HitSystems/tocBackend/compare/v3.1.5...v3.1.6) (2022-04-01)

### [3.1.5](https://github.com/HitSystems/tocBackend/compare/v3.1.4...v3.1.5) (2022-03-31)

### [3.1.4](https://github.com/HitSystems/tocBackend/compare/v3.1.3...v3.1.4) (2022-03-25)

### [3.1.3](https://github.com/HitSystems/tocBackend/compare/v3.1.2...v3.1.3) (2022-03-21)

### [3.1.2](https://github.com/HitSystems/tocBackend/compare/v3.1.1...v3.1.2) (2022-03-15)

### [3.1.1](https://github.com/HitSystems/tocBackend/compare/v3.0.3...v3.1.1) (2022-03-15)

### [3.0.28](https://github.com/HitSystems/tocBackend/compare/v3.0.27...v3.0.28) (2022-02-21)

### [3.0.27](https://github.com/HitSystems/tocBackend/compare/v3.0.26...v3.0.27) (2022-02-18)

### [3.0.26](https://github.com/HitSystems/tocBackend/compare/v3.0.25...v3.0.26) (2022-02-16)

### [3.0.25](https://github.com/HitSystems/tocBackend/compare/v3.0.24...v3.0.25) (2022-02-15)

### [3.0.24](https://github.com/HitSystems/tocBackend/compare/v3.0.23...v3.0.24) (2022-02-07)

### [3.0.23](https://github.com/HitSystems/tocBackend/compare/v3.0.22...v3.0.23) (2022-02-03)

### [3.0.22](https://github.com/HitSystems/tocBackend/compare/v3.0.21...v3.0.22) (2022-02-01)

### [3.0.21](https://github.com/HitSystems/tocBackend/compare/v3.0.20...v3.0.21) (2022-01-31)

### [3.0.20](https://github.com/HitSystems/tocBackend/compare/v3.0.19...v3.0.20) (2022-01-26)

### [3.0.19](https://github.com/HitSystems/tocBackend/compare/v3.0.18...v3.0.19) (2022-01-22)

### [3.0.18](https://github.com/HitSystems/tocBackend/compare/v3.0.17...v3.0.18) (2022-01-20)

### [3.0.17](https://github.com/HitSystems/tocBackend/compare/v3.0.16...v3.0.17) (2022-01-20)

### [3.0.16](https://github.com/HitSystems/tocBackend/compare/v3.0.15...v3.0.16) (2022-01-20)

### [3.0.15](https://github.com/HitSystems/tocBackend/compare/v3.0.14...v3.0.15) (2022-01-12)

### [3.0.14](https://github.com/HitSystems/tocBackend/compare/v3.0.13...v3.0.14) (2022-01-11)

### [3.0.13](https://github.com/HitSystems/tocBackend/compare/v3.0.12...v3.0.13) (2022-01-02)

### [3.0.12](https://github.com/HitSystems/tocBackend/compare/v3.0.11...v3.0.12) (2021-12-23)

### [3.0.11](https://github.com/HitSystems/tocBackend/compare/v3.0.10...v3.0.11) (2021-12-21)

### [3.0.10](https://github.com/HitSystems/tocBackend/compare/v3.0.9...v3.0.10) (2021-12-18)

### [3.0.9](https://github.com/HitSystems/tocBackend/compare/v3.0.8...v3.0.9) (2021-12-17)

### [3.0.8](https://github.com/HitSystems/tocBackend/compare/v3.0.7...v3.0.8) (2021-12-16)

### [3.0.7](https://github.com/HitSystems/tocBackend/compare/v3.0.6...v3.0.7) (2021-12-14)

### [3.0.6](https://github.com/HitSystems/tocBackend/compare/v3.0.4...v3.0.6) (2021-12-10)

### [3.0.5](https://github.com/hitsystems/tocBackend/compare/v3.0.3...v3.0.5) (2021-12-03)

### [3.0.4](https://github.com/HitSystems/tocBackend/compare/v3.0.3...v3.0.4) (2021-12-03)

### [3.0.3](https://github.com/HitSystems/tocBackend/compare/v3.0.2...v3.0.3) (2021-12-03)

### [3.0.2](https://github.com/HitSystems/tocBackend/compare/v3.0.1...v3.0.2) (2021-12-03)

### 3.0.1 (2021-12-03)


src/sanPedro.ts
import {ticketsInstance} from './tickets/tickets.clase';
import {sincronizarTickets, sincronizarCajas, sincronizarMovimientos, sincronizarFichajes, sincronizarDevoluciones} from './sincro';
import {cajaInstance} from './caja/caja.clase';
import {movimientosInstance} from './movimientos/movimientos.clase';
import {trabajadoresInstance} from './trabajadores/trabajadores.clase';
import {devolucionesInstance} from './devoluciones/devoluciones.clase';
import { logger } from "./logger";

const io = require('socket.io-client');
const socket = io('https://sanpedro.cloud');
// const socket = io('http://localhost:3001'); // DEV SANPEDRO EN LOCAL


function emitSocket(canal: string, datos: any = null) {
  if (socket.connected) {
    socket.emit(canal, datos);
  }
}

socket.on("resSincroTickets", async (data) => {
  if (data.error == false) {
    if (data.ticket) {
      if (await ticketsInstance.actualizarEstadoTicket(data.ticket)) {
        sincronizarTickets(true);
      } else {
        logger.Error(19, "Error al actualizar el ticket");
      }
    }
  } else {
    if (typeof data.ticket.comentario == "string") {
      if (data.mensaje == "SanPedro: Error, parámetros incorrectos") {
        data.ticket.comentario = "SanPedro: Error, parámetros incorrectos";
      }
    }
  }
});

socket.on('resCajas', (data) => {
  if (data.error == false) {
    if (data.repetir == false) {
      cajaInstance.confirmarCajaEnviada(data.infoCaja).then((res) => {
        if (res) {
          sincronizarCajas();
        } else {
          logger.Error(20, 'Error al actualizar el estado de la caja');
        }
      }).catch((err) => {
        logger.Error(21, err);
      });
    } else {
      cajaInstance.confirmarCajaEnviada(data.infoCaja).then((res) => {
        if (res) {
          sincronizarCajas();
        } else {
          logger.Error(22, 'Error al actualizar el estado de la caja 2');
        }
      }).catch((err) => {
        logger.Error(23, err);
      });
      // cambiar estado infoCaja en mongo (enviado + comentario)
    }
  } else {
    logger.Error(24, data.mensaje);
  }
});

socket.on('resMovimientos', (data) => {
  if (data.error == false) {
    movimientosInstance.actualizarEstadoMovimiento(data.movimiento).then((res) => {
      if (res) {
        sincronizarMovimientos(true);
      } else {
        logger.Error(25, 'Error al actualizar el estado del movimiento');
      }
    }).catch((err) => {
      logger.Error(26, err);
    });
  } else {
    logger.Error(27, data.mensaje);
  }
});

socket.on('resFichajes', (data) => {
  if (data.error == false) {
    trabajadoresInstance.actualizarEstadoFichaje(data.fichaje).then((res) => {
      if (res) {
        sincronizarFichajes();
      } else {
        logger.Error(28, 'Error al actualizar el estado del fichaje');
      }
    }).catch((err) => {
      logger.Error(29, err);
    });
  } else {
    logger.Error(30, data.mensaje);
  }
});

socket.on('resSincroDevoluciones', (data) => {
  if (!data.error) {
    devolucionesInstance.actualizarEstadoDevolucion(data.devolucion).then((res) => {
      if (res) {
        sincronizarDevoluciones();
      } else {
        logger.Error(31, 'Error al actualizar el estadio de la devolución.');
      }
    }).catch((err) => {
      logger.Error(32, err);
    });
  } else {
    logger.Error(33, data.mensaje);
  }
});

export {socket, emitSocket};
